rising/transforms/affine.py

```diff
@@ -252,7 +210,226 @@ def __radd__(self, other):
+            assemble stacked matrices. Afterwards these transformations are
+            stacked by matrix-multiplication to only perform a single
+            interpolation
+        keys: Sequence
+            keys which should be augmented
+        grad: bool
+            enable gradient computation inside transformation
+        output_size : Iterable
+            if given, this will be the resulting image size.
+            Defaults to ``None``
+        adjust_size : bool
+            if True, the resulting image size will be calculated dynamically
+            to ensure that the whole image fits.
+        interpolation_mode : str
+            interpolation mode to calculate output values
+            'bilinear' | 'nearest'. Default: 'bilinear'
+        padding_mode :
+            padding mode for outside grid values
+            'zeros' | 'border' | 'reflection'. Default: 'zeros'
+        align_corners : Geometrically, we consider the pixels of the input as
+            squares rather than points. If set to True, the extrema (-1 and 1)
+            are considered as referring to the center points of the input’s
+            corner pixels. If set to False, they are instead considered as
+            referring to the corner points of the input’s corner pixels,
+            making the sampling more resolution agnostic.
+        **kwargs :
+            additional keyword arguments passed to the affine transform
+
+        """
+
+        if isinstance(transforms, (tuple, list)):
+            if isinstance(transforms[0], (tuple, list)):
+                transforms = transforms[0]
+
+        # ensure trafos are Affines and not raw matrices
+        transforms = tuple(
+            [trafo if isinstance(trafo, Affine) else Affine(matrix=trafo)
+             for trafo in transforms])
+
+        super().__init__(keys=keys, grad=grad,
+                         output_size=output_size, adjust_size=adjust_size,
+                         interpolation_mode=interpolation_mode,
+                         padding_mode=padding_mode,
+                         align_corners=align_corners,
+                         **kwargs)
+
+        self.transforms = transforms
+
+    def assemble_matrix(self, **data) -> torch.Tensor:
+        """
+        Handles the matrix assembly and stacking
+
+        Parameters
+        ----------
+        **data :
+            the data to be transformed. Will be used to determine batchsize,
+            dimensionality, dtype and device
+
+        Returns
+        -------
+        torch.Tensor
+            the (batched) transformation matrix
+
+        """
+        whole_trafo = None
+        for trafo in self.transforms:
+            matrix = matrix_to_homogeneous(trafo.assemble_matrix(**data))
+            if whole_trafo is None:
+                whole_trafo = matrix
+            else:
+                whole_trafo = torch.bmm(whole_trafo, matrix)
+        return matrix_to_cartesian(whole_trafo)
+
+
+class BaseAffine(Affine):
```

Comment:
Why did you decide to outsource this to a separate class? IMO a complete class may be overkill for just this one function. Couldn't just include it in the main `Affine`?

Reply:
I like the single responsibility principle :) Furthermore, this will help us in the future. With this outsourcing it is possible for us to add new representations in the future without rewriting the base class. The BaseAffine is just one special case how one can visualise transformations, e.g. rotation can also be modelled by an rotation around a parametrised axis (rotate 50 deg around [1,1,1])